cmd/mdatagen/internal/metadata.go

```diff
@@ -0,0 +1,288 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/collector/cmd/mdatagen/internal"
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"go.opentelemetry.io/collector/filter"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+type Metadata struct {
+	// Type of the component.
+	Type string `mapstructure:"type"`
+	// Type of the parent component (applicable to subcomponents).
+	Parent string `mapstructure:"parent"`
+	// Status information for the component.
+	Status *Status `mapstructure:"status"`
+	// The name of the package that will be generated.
+	GeneratedPackageName string `mapstructure:"generated_package_name"`
+	// Telemetry information for the component.
+	Telemetry Telemetry `mapstructure:"telemetry"`
+	// SemConvVersion is a version number of OpenTelemetry semantic conventions applied to the scraped metrics.
+	SemConvVersion string `mapstructure:"sem_conv_version"`
+	// ResourceAttributes that can be emitted by the component.
+	ResourceAttributes map[AttributeName]Attribute `mapstructure:"resource_attributes"`
+	// Attributes emitted by one or more metrics.
+	Attributes map[AttributeName]Attribute `mapstructure:"attributes"`
+	// Metrics that can be emitted by the component.
+	Metrics map[MetricName]Metric `mapstructure:"metrics"`
+	// GithubProject is the project where the component README lives in the format of org/repo, defaults to open-telemetry/opentelemetry-collector-contrib
+	GithubProject string `mapstructure:"github_project"`
+	// ScopeName of the metrics emitted by the component.
+	ScopeName string `mapstructure:"scope_name"`
+	// ShortFolderName is the shortened folder name of the component, removing class if present
+	ShortFolderName string `mapstructure:"-"`
+	// Tests is the set of tests generated with the component
+	Tests Tests `mapstructure:"tests"`
+}
+
+func (md *Metadata) Validate() error {
+	var errs error
+	if err := md.validateType(); err != nil {
+		errs = errors.Join(errs, err)
+	}
+
+	if md.Parent != "" {
+		if md.Status != nil {
+			// status is not required for subcomponents.
+			errs = errors.Join(errs, errors.New("status must be empty for subcomponents"))
+		}
```

Comment:
It looks like this error condition was not present on the original file, and it is making the contrib release fail. Part of the solution is #11167, but could we also have these kinds of changes listed in the PR description (and maybe changelog as well)? It was a bit hard for me to debug this (since the PR title talks about just moving files around, I discarded this PR as the cause of the error initially)

Reply:
I vote to reset back to the original logic, and have an issue if we want to evolve the logic.